dso-l2/src/main/java/com/tc/l2/state/ServerVoterManagerImpl.java

```diff
@@ -67,7 +67,7 @@ public int getVoterLimit() {
 
   @Override
   public synchronized long registerVoter(String id) {
-    if (voters.containsKey(id)) {
+    if (!votingInProgress && voters.containsKey(id)) {
       //  already registered.  double register is not supported
```

Comment:
I don't understand this change.  HEARTBEAT_RESPONSE means that the voter is already registered.  I don't think timeout needs to be considered but maybe.